### PR TITLE
fix: Adds a set transformation before returning the dynamicbindingpathlist to ensure that dulplicates don't go through

### PR DESCRIPTION
**Pull request by rahulbarwal**
Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!CAUTION]
> 🔴 🔴 🔴 Some tests have failed.
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11105992104>
> Commit: 9a88d592a2577bbf0618c8d35b436796bf3aa0ca
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11105992104&attempt=1&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank">Cypress dashboard</a>.
> Tags: @tag.All
> Spec: 
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Regression/ClientSide/Widgets/Button/Button_showAlert_multiline_message_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Widgets/Chart/ChartDataPoint_Spec.ts
> <li>cypress/e2e/Regression/ClientSide/Widgets/Dropdown/Dropdown_onOptionChange_spec.js
> <li>cypress/e2e/Regression/ClientSide/Widgets/TableV1/Table_Widget_Add_button_spec.js
> <li>cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_2_spec.js
> <li>cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_3_spec.js
> <li>cypress/e2e/Regression/ClientSide/Widgets/TableV2/TableV2_Widget_Add_button_spec.js
> <li>cypress/e2e/Regression/ClientSide/Widgets/TableV2/inline_editing_validations_spec.js</ol>
> <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">List of identified flaky tests</a>.
> <hr>Mon, 30 Sep 2024 13:38:14 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced uniqueness of dynamic binding paths to improve data integrity.
  
- **Bug Fixes**
	- Improved safety checks in utility functions to prevent errors when accessing properties on undefined or non-array values.

### CODE REVIEW

File: app/client/src/utils/DynamicBindingUtils.ts [hunk 1]
```diff
@@ -193,7 +193,7 @@ export const getEntityDynamicBindingPathList = (
     entity.dynamicBindingPathList &&
     Array.isArray(entity.dynamicBindingPathList)
   ) {
-    return [...entity.dynamicBindingPathList];
+    return [...Array.from(new Set(entity.dynamicBindingPathList))];
```
Comment (rishabhrathod01) [inline app/client/src/utils/DynamicBindingUtils.ts:196]: Could we skip using `Array.from` here? A set can be spread directly to an array like `[...set]`

Comment (rahulbarwal) [reply app/client/src/utils/DynamicBindingUtils.ts]: Done.

**Comment by rahulbarwal** [comment]
closing as no longer required.

**Pull request closed without merging**